test/tests/functional/pbs_hook_set_jobenv.py

```diff
@@ -180,32 +182,52 @@ def common_log_match(self, daemon):
                   "TEST_SQUOTE5=music \\'makes \\\"the\\'\\\"people",
                   "TEST_SQUOTE6=loving\\'",
                   "TEST_SPECIAL={}[]()~@#$%^&*!",
-                  "TEST_SPECIAL2=<dumb-test_text>"]
+                  "TEST_SPECIAL2=<dumb-test_text>",
+                  "TEST_RETURN=\\'3\,",
+                  # Cannot add '\n' here because '\n' is not included in
+                  # the items of the list returned by log_lines(), (though
+                  # lines are split by '\n')
+                  "4\,",
+                  "5\\',"]
 
         if (daemon == "mom"):
             self.logger.info("Matching in mom logs")
+            logfile_type = self.mom
         elif (daemon == "server"):
             self.logger.info("Matching in server logs")
+            logfile_type = self.server
         else:
             self.logger.info("Provide a valid daemon name; server or mom")
             return
-
+        lines = None
+        ret_linenum = 0
+        search_msg = 'log match: searching for '
+        nomatch_msg = ' No match for '
         for msg in logmsg:
-            if (daemon == "mom"):
-                # Match the trailing separator (',')
-                self.mom.log_match(msg + ',')
-            elif (daemon == "server"):
-                self.server.log_match(msg, starttime=self.server.ctime)
-
-        # Following lines are commented due to PTL bug PP-1008
-        # if (daemon == "mom"):
-        #    rv = self.mom.log_match(
-        #          msg="TEST_RETURN\=\\\'3\\\,\n4\\\,\n5\\\'", regexp=True)
-        #    self.assertTrue(rv)
-        # else:
-        #    rv = self.server.log_match(
-        #            msg="TEST_RETURN\=\\\'3\\\,\n4\\\,\n5\\\'", regexp=True)
-        #    self.assertTrue(rv)
+            attempt = 1
+            while attempt <= 60:
```

Comment:
@bhroam , Thanks for reviewing this.   Yes. You are correct. The fix here is a workaround due to log_match's current behavior. 
There is a ticket already filed for fixing log_match itself (PP-1008).  But, two reasons why I preferred doing this : 
a) I don't see any other use cases where a test would want to log_match a new line without a timestamp.  As far I looked up, this is the only test that got affected by log_match's current deficiency.  
b) Changing the behavior of log_match would be a bigger change and requires more validation.  Fixing the test is very local.
Your thoughts? If you still feel it would be be better to fix log_match itself, we could do that as well. Thanks

Reply:
> You are iterating here, I'd use a for _ in range(60): loop

Wil fix this